README.md

```diff
@@ -1,6 +1,98 @@
+```bash
+pre-commit install
+```
+This will make sure your commits will be accepted by the CI Pipeline.
+
+## Unit-Tests
+Unit-tests are divided into two categories:
+
+- Application or Integration tests: These tests make sure, the application functions properly.
+- Extractor tests: These tests test the individual feature generation units.
+
+All tests make use of mocks and static (checked in) resources. Hence, they can run independently of any docker
+container or other environment constraints.
+
+## Branching Model
+This repository follows the [GitHub Flow](http://scottchacon.com/2011/08/31/github-flow.html) branching model:
+
+> - Anything in the main branch is deployable
+> - To work on something new, create a descriptively named branch off of master (ie: new-oauth2-scopes)
+> - Commit to that branch locally and regularly push your work to the same named branch on the server
+> - When you need feedback or help, or you think the branch is ready for merging, open a pull request
+> - After someone else has reviewed and signed off on the feature, you can merge it into main
+> - Once it is merged and pushed to ‘main’, you can and should deploy immediately
+
+## Definition of Done
+- The service is able to perform with just the url of the website. No additional information is needed.
+- The service is resilient against failing code, i.e., the user will always get a response.
+  In the worst case, that response is an error message.
+- CI Pipeline passes (Unit tests, pre-commit, and build steps)
+
+## CI Pipeline
+The pipeline is launched by creating or updating a branch and has to succeed before any pull request can be merged.
+It will perform the following steps:
+ - run pre-commit hooks to validate codestyle and quality
+ - build python package
+ - run python unittests
+ - build docker images
+ - (on main branch) push docker images to registry.
+
+## Releases & Versioning
+- This repository uses [Sem-Ver](https://semver.org/lang/de/).
+- New releases are created via the GitHub release interface.
+- Release Tags must follow the pattern `v{MAJOR}.{MINOR}.{PATCH}`
+- Docker tags will be extracted from the git tags as `{MAJOR}.{MINOR}.{PATCH}`
+
+## Deployment
+MetaLookup requires Splash and Lighthouse containers to be running. To deploy them together the two docker-compose files
+for [dev](./meta-lookup-compose-dev.yml) and [prod](./meta-lookup-compose-prod.yml) are used.
+
+## Configuration
+The application can be configured via environment variables or a `.env` file. See
+[`settings.py`](./src/metalookup/lib/settings.py) for the available options.
+
+## Health-Checks
+The service provides a `/_ping` endpoint which will respond with a `200 OK` (body: `{"status": "ok"}`) immediately
+(within a couple of milliseconds). If the request takes to long, or answers differently, then the service is in an
+inconsistent state and should be restarted. The docker-compose files use this endpoint for health checks and automated
```

Comment:
I do not believe the docker-compose files do the health checks - I presume that is the docker demon and potentially the service mesh, e.g., zabbix.

Reply:
its the daemon, rephrased!